keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1256 @@
+      selectors:
+      - cel:
+          expression: "device.attributes['vlan-cni.networking.x-k8s.io'].name == 'eth1'"
+      resources:
+        requests:
+          bandwidth: "1Gi"
+```
+
+#### Story 4
+A user defines more than one requests to get devices from different sources. From the following claim spec, the claim should select two devices which have different device sources.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: net1
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      resources:
+        bandwidth: "1Gi"
+    - name: net2
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      resources:
+        requests:
+          bandwidth: "1Gi"
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+This enhancement introduces the `consumableCapacity` field within the `BasicDevice` of the ResourceSlice. 
+The consumableCapacity can be set to infinity.
+Devices that have at least one consumableCapacity are considered to be shared devices.
+
+Users can only claim a shared device using the newly added `Shared` allocation mode (`DeviceAllocationMode`). In other words, the shared device will be skipped in all other modes, while the unsharable device will be skipped in the Shared mode.
+
+Users can define specific per-device resource requests through
+the newly added resources field in the `DeviceRequest` of the `ResourceClaim`. 
+If the resources field is not specified, all consumable resources are assumed.
+
+A shared device will only be allocated once its consumability has been verified
+and its attributes match the request's selectors and constraints.
+"share" refers to amount of resources reserved or allocated to the claim request.
+The newly added `allocatedShare` field in the `DeviceRequestAllocationResult` will be set when the allocation is successful.
+A device can be shared (allocated) to different pods. 
+However, claims from the same pod cannot allocate shares from the same shared device.
+
+### API enhancement
+To enable this enhancement, the following API updates are proposed.
+#### ResourceSliceSpec's BasicDevice
+
+```go
+type BasicDevice struct {
+	Attributes         map[QualifiedName]DeviceAttribute
+	Capacity           map[QualifiedName]DeviceCapacity
+	ConsumableCapacity map[QualifiedName]DeviceConsumableCapacity
```

Comment:
how does both maps are related? are capacities allowed to be on both places?

Reply:
Currently, the value in capacity is not used in allocation decision. I'm not sure the original purpose of the capacity so that I define a new field for consumable capacity to be used in allocation decision.

Reply:
The `capacity` field specifies the available capacities for the device, so my intuition here is that we don't need a new `ConsumableCapacity` field. With the introduction of the `ConsumesCapacity` field in https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4815-dra-partitionable-devices, we make a distinction between the capacity a device consumes from the internal pool and the advertised capacity. Having three fields named `Capacity`, ConsumesCapacity` and `ConsumableCapacity` will probably get somewhat confusing.